allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/interceptors/testrun/MoveReportsInterceptor.kt

```diff
@@ -0,0 +1,133 @@
+package com.kaspersky.components.alluresupport.interceptors.testrun
+
+import android.app.Instrumentation
+import androidx.test.uiautomator.UiDevice
+import com.google.common.io.CharStreams
+import com.kaspersky.components.alluresupport.files.resources.AllureResourcesRootDirsProvider
+import com.kaspersky.kaspresso.files.dirs.DirsProvider
+import com.kaspersky.kaspresso.interceptors.watcher.testcase.TestRunWatcherInterceptor
+import com.kaspersky.kaspresso.testcases.models.info.TestInfo
+import io.qameta.allure.kotlin.Allure
+import io.qameta.allure.kotlin.util.PropertiesUtils
+import org.json.JSONObject
+import java.io.File
+
+private const val ATTACHMENTS_JSON_FIELD = "attachments"
+private const val NAME_JSON_FIELD = "name"
+private const val SOURCE_JSON_FIELD = "source"
+private const val MP4_EXTENSION = "mp4"
+
+/**
+ * Current allure version stores reports to /data/data/your.package.name/files/allure-results.
+ * This interceptor moves them to default artifacts folder after test and replaces mock videos with real ones
+ */
+class MoveReportsInterceptor(
+    private val instrumentation: Instrumentation,
+    private val dirsProvider: DirsProvider,
+    private val rootDirsProvider: AllureResourcesRootDirsProvider,
+    private val videosHolder: AttachedAllureVideosHolder,
+    private val device: UiDevice
+) : TestRunWatcherInterceptor {
+
+    private var lastTestCaseUuid: String? = null
+
+    override fun onMainSectionStarted(testInfo: TestInfo) {
+        lastTestCaseUuid = Allure.lifecycle.getCurrentTestCase()
+    }
+
+    override fun onTestFinished(testInfo: TestInfo, success: Boolean) {
+        val allureTargetDir = moveAllureReportToSdCard()
+        removeStubs(allureTargetDir)
+        moveAttachedVideosToProperDirectories(allureTargetDir)
+        cleanUp()
+    }
+
+    /**
+     * Deletes allure report under /data/data/your.package.name/files and empty directories after moving real videos from them
+     */
+    private fun cleanUp() {
+        getOriginalAllureDir().deleteRecursively()
+        val videosDir = dirsProvider.provideNew(rootDirsProvider.videoRootDir)
+        device.executeShellCommand("rm -rf ${videosDir.absolutePath}")
+
+        dirsProvider.provideNew(rootDirsProvider.screenshotsRootDir).deleteRecursively()
```

Comment:
This would be a breaking change. Not sure that we have to clear this, previously these files were just copied to the report. I would say we can leave it as it was, but not insisting, need more opinions here. At least it should be mentioned in the changelog.